### ISSUE TITLE
AzureML: Execute YAML mangling file paths in Git Bash

### ISSUE THREAD

**Issue by adriantorrie** [microsoft/vscode-tools-for-ai#1844]
## Expected Behavior

- Azure ML Environment is created.

## Actual Behavior

- File path to conda definition is mangled

```sh
$ az ml environment create --subscription XXXXXX --resource-group XXXXXX --workspace-name XXXXXX --file c:\Users\torriea\repos\scaffold_model_python\src\assets\environment_dev.yml


Error: The yaml file you provided does not match the prescribed schema for General yaml files and/or has the following issues:

1) One or more files or folders do not exist.


Details:

No such file or directory: c:Userstorrieareposscaffold_model_pythonsrcassetsenvironment_dev.yml

Resolutions:
Double-check the directory paths you provided and enter the correct paths.
If using the CLI, you can also check the full log in debug mode for more details by adding --debug to the end of your command
Additional Resources: The easiest way to author a yaml specification file is using IntelliSense and auto-completion Azure ML VS code extension provides: https://code.visualstudio.com/docs/datascience/azure-machine-learning. To set up VS Code, visit https://docs.microsoft.com/azure/machine-learning/how-to-setup-vs-code
```

## Steps to Reproduce the Problem

  1. VSCode configured with Git Bash as the default terminal
  2. Right click YAML file
  1. Select `AzureML: Execute YAML`
  1. Error occurs as above

## Specifications

  - Version: 
![image](https://user-images.githubusercontent.com/3909039/207990229-60036db7-44cd-496d-9ef4-c18843065e0e.png)

  - Platform: Windows

**Comment by shsuman** [comment]
@adriantorrie What terminal are you using ? Can you try it in Command Prompt ?

As far as I can tell, the extension is executing the correct command but somewhere in the middle, either because of some weird escape sequence of the terminal or the ml cli extension, the backslashes are ignored

**Comment by adriantorrie** [comment]
Git Bash is the default terminal.

When parsing the file path, the windows style path is having the backslashes removed, as on bash they need to be escaped.

Easy fix is to have the extension quote the file path.

**Comment by shsuman** [comment]
@adriantorrie I can repro it as well. This seems to be a problem with all bash terminals on windows. Until now, we only quoted file paths if there was a space character in it but now it seems like we need to use quotes for all case just to be safe :).

For the time being, please switch to a cmd or power shell. We will try to fix it in the next release

**Comment by shsuman** [comment]
@adriantorrie I have put out a fix microsoft/vscode-azureml#344 for this. I will let you know once its in pre release :)

**Comment by adriantorrie** [comment]
> @adriantorrie I have put out a fix microsoft/vscode-azureml#344 for this. I will let you know once its in pre release :)

Awesome, thank you.
